src/dune_pkg/package_variable.ml

```diff
@@ -63,20 +63,33 @@ let to_macro_invocation { name; scope } =
 
 let to_pform t = Pform.Macro (to_macro_invocation t)
 
-let of_opam_ident ident =
-  match String.lsplit2 ident ~on:':' with
-  | Some ("_", variable) -> `Package_variable (self_scoped (Name.of_string variable))
-  | Some (package, variable) ->
+let package_variable package variable =
+  match package, variable with
+  | _, ("root" | "hash" | "build-id" | "misc") -> `Unsupported_variable variable
```

Comment:
So we're only excluding unsupported variables inside the translation, but not inside the substitution, right?

Reply:
I think so. That's something #8715 shows. I'll reconfirm after this PR is merged.